sklearn/metrics/_classification.py

```diff
@@ -1758,10 +1758,14 @@ def precision_recall_fscore_support(
         # score = (1 + beta**2) * precision * recall / (beta**2 * precision + recall)
         # We set to `zero_division_value` if the denominator is 0 **or** if **both**
         # precision and recall are ill-defined.
+        numer = (1 + beta2) * precision * recall
         denom = beta2 * precision + recall
-        mask = np.isclose(denom, 0) | np.isclose(pred_sum + true_sum, 0)
-        denom[mask] = 1  # avoid division by 0
-        f_score = (1 + beta2) * precision * recall / denom
+        denom_mask = np.isclose(denom, 0)
+        mask = (np.not_equal(numer, 0) & np.isclose(denom, 0)) | np.isclose(
+            pred_sum + true_sum, 0
+        )
+        denom[denom_mask] = 1  # avoid division by 0
+        f_score = numer / denom
```

Comment:
Considering the issue reported in #27189, I think that we can express it without relying on the precision and recall:

```python
# The score is defined as:
        # score = (1 + beta**2) * precision * recall / (beta**2 * precision + recall)
        # Therefore, you can express the score in terms of confusion matrix entries as:
        # score = (1 + beta**2) * tp / ((1 + beta**2) * tp + beta**2 * fn + fp)
        denom = beta2 * true_sum + pred_sum
        f_score = _prf_divide(
            (1 + beta2) * tp_sum,
            denom,
            "fscore",
            "true nor predicted",
            average,
            warn_for,
            zero_division,
        )
```

In this case, we handle it as any other by zero division.